routing/endpointregistry.go

```diff
@@ -14,37 +15,62 @@ const defaultLastSeenTimeout = 1 * time.Minute
 type Metrics interface {
 	DetectedTime() time.Time
 	InflightRequests() int64
+	IncInflightRequest()
+	DecInflightRequest()
 }
 
 type entry struct {
-	detected         time.Time
-	inflightRequests int64
+	detected         atomic.Value // time.Time
+	lastSeen         atomic.Value // time.Time
+	inflightRequests atomic.Int64
 }
 
 var _ Metrics = &entry{}
 
 func (e *entry) DetectedTime() time.Time {
-	return e.detected
+	return e.detected.Load().(time.Time)
```

Comment:
This can return nil if the time was not stored before

Reply:
That's why I always call `LoadOrStore` method with `newEntry()` argument which ensures that `e.detected` is always `time.Time`.

Reply:
Maybe I can add some unit test(s) to ensure this type conversation will never panic.